api/src/main/java/edu/cornell/mannlib/vitro/webapp/modules/messaging/JMSMessagingClient.java

```diff
@@ -0,0 +1,108 @@
+package edu.cornell.mannlib.vitro.webapp.modules.messaging;
+
+import java.util.Properties;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.cornell.mannlib.vitro.webapp.modules.Application;
+import edu.cornell.mannlib.vitro.webapp.modules.ComponentStartupStatus;
+import edu.cornell.mannlib.vitro.webapp.utils.configuration.Property;
+
+public class JMSMessagingClient {
+
+    private static final Log log = LogFactory.getLog(JMSMessagingClient.class);
+
+    private String factoryInitial;
+
+    private String providerURL;
+
+    private String connectionFactory;
+
+    private String brokerDestination;
+
+    private String brokerUsername;
+
+    private String brokerPassword;
+
+    private Connection connection;
```

Comment:
It does not appear that `connection` (or `destination`, below) need to be class members. They could probably be local variables in `startup()`.